src/lib/operators/jit_operator/operators/jit_filter.cpp

```diff
@@ -14,7 +14,7 @@ std::shared_ptr<const JitExpression> JitFilter::expression() { return _expressio
 
 void JitFilter::_consume(JitRuntimeContext& context) const {
   const auto result = _expression->compute<bool>(context);
-  if ((!_expression->result_entry().is_nullable() || result.has_value()) && result.value()) {
```

Comment:
Question here, which is not directly related to your change: if `!_expression->result_entry().is_nullable()` is True, then `&& result.value()` will be called no matter if `result.has_value()` is True or False, correct? And in that case, it throws an exception. Shouldn't this if statement require the check for `result.has_value()` (or rather, just `result` with your changes) in any case?

This comes up in a few other places as well.

Reply:
@FabianWiebe this is JIT, maybe you know if the `if` is doing what it intends to?

Yes, `result.value()` may be called even with `result.has_value()`. Unless there is a dependency that makes sure that `!_expression->result_entry().is_nullable() -> result.has_value()`...

Reply:
@lanice If a value is not nullable, the optional is always set so that the` has_value()` check is not necessary.